### PR TITLE
[TwigBunde] [WebProfilerBundle] Exclude Some Vendors From Being Hidden

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature? | yes 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #27409   
| License       | MIT
| Doc PR        | todo when implementation is validated

This implementation adds a new parameter to allow excluded vendor.

### CODE REVIEW

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ExceptionController.php [hunk 1]
```diff
@@ -31,13 +31,15 @@ class ExceptionController
...
+    private $privateVendorsList;
 
-    public function __construct(Profiler $profiler = null, Environment $twig, bool $debug, FileLinkFormatter $fileLinkFormat = null)
+    public function __construct(Profiler $profiler = null, Environment $twig, bool $debug, FileLinkFormatter $fileLinkFormat = null, array $privateVendorsList)
     {
         $this->profiler = $profiler;
         $this->twig = $twig;
         $this->debug = $debug;
         $this->fileLinkFormat = $fileLinkFormat;
+        $this->privateVendorsList = $privateVendorsList;
```
Comment (OskarStark) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ExceptionController.php]: To me the “List” is superfluous, what do you think?

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php [hunk 2]
```diff
@@ -47,13 +47,13 @@ public function load(array $configs, ContainerBuilder $container)
 
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
         $loader->load('profiler.xml');
-
```
Comment (OskarStark) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php]: I am for keeping this line for readability

**Comment by Simperfit** [comment]
Thanks @OskarStark for the review !

**Review by OskarStark — APPROVED** [review]

File: src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php [hunk 3]
```diff
@@ -54,6 +54,7 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setParameter('web_profiler.debug_toolbar.intercept_redirects', $config['intercept_redirects']);
             $container->setParameter('web_profiler.debug_toolbar.mode', $config['toolbar'] ? WebDebugToolbarListener::ENABLED : WebDebugToolbarListener::DISABLED);
         }
+        $container->setParameter('web_profiler.debug_toolbar.private_vendors', $config['private_vendors']);
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/DependencyInjection/WebProfilerExtension.php]: why not inject the argument directly?

File: src/Symfony/Bundle/TwigBundle/Resources/views/Exception/traces.html.twig [hunk 4]
```diff
@@ -22,7 +22,7 @@
         <div id="trace-html-{{ index }}" class="sf-toggle-content">
         {% set _is_first_user_code = true %}
         {% for i, trace in exception.trace %}
-            {% set _is_vendor_trace = trace.file is not empty and ('/vendor/' in trace.file or '/var/cache/' in trace.file) %}
+            {% set _is_vendor_trace = trace.file is not empty and ('/vendor/' in trace.file or '/var/cache/' in trace.file or trace.file not in private_vendors) %}
```
Comment (ro0NL) [inline src/Symfony/Bundle/TwigBundle/Resources/views/Exception/traces.html.twig]: `trace.file not in private_vendors`

this requires the abs. file path to be configured? :thinking: shouldnt we test each private vendor separate?

File: src/Symfony/Bundle/WebProfilerBundle/Controller/ExceptionController.php [hunk 5]
```diff
@@ -75,6 +77,7 @@ public function showAction($token)
                 'status_text' => Response::$statusTexts[$code],
                 'exception' => $exception,
                 'logger' => null,
+                'private_vendors' => $this->privateVendors,
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Controller/ExceptionController.php]: twig-bundle dependency needs a bump to 4.3

**Review by ro0NL — COMMENTED** [review]
last but not least, does this enforce the same behavior between the exception page in the profiler and outside? should we find a different configuration hook?

**Comment by nicolas-grekas** [comment]
The linked issue suggests:

>  When vendor part of PSR-4 namespace matches current project namespace code should be treated as belonging to the project..

it would be great to try implementing this instead.

As far as this PR is concerned, I don't think adding an option for this will work. Almost nobody has time to discover, learn, configure, maintain such a feature that "only" enhances DX marginally.

**Comment by Simperfit** [comment]
I've taken @nicolas-grekas comments in account, just taking some time to think on how to do this, will repush in this PR lately in the week.

**Comment by Simperfit** [comment]
Status: Needs Work

**Pull request closed without merging**